lib/fs/readlink/readlinknul.h

```diff
@@ -43,7 +43,7 @@ readlinknul(const char *restrict link, char *restrict buf, size_t size)
 	}
 
 	stpcpy(&buf[len], "");
-	return len;
+	return 0;
```

Comment:
> We don't use the length at call site.  Just return 0 for success.

But the comment above the fn says:

```
 29 // Similar to readlink(2), but terminate the string.
```

so I think it should make sure to stick to the 'RETURN VALUE' section of readlink(2) manpage.

Reply:
Hmmm, yeah, that.  I internally agree.  I just ditched the value in an attempt to get rid of sign-mismatch warnings.  I'll try to get rid of them while keeping the return value.

Reply:
Oh, actually I can just drop commit 2.

Reply:
Ahhhh, I now remember!  I did it because while writing the function I had two conflicting ideas in my head:

-  readlink(3) unnecessarily returns the length.  Who needs it?  Let's keep it simple and return 0/-1 in an int.
-  readlink(3) returns the length or an error code, in a ssize_t.

In the end, I got a brain fart and did a mix.  It returns the length or an error code via int.  So I had to either return 0/-1 or a ssize_t.  And opted for the former.  I guess I can do the latter, and change the return to be a ssize_t.